packages/core/src/connectors/alipay/index.ts

```diff
@@ -0,0 +1,243 @@
+const pathToConfigTemplate = path.join(currentPath, 'config-template.md');
+const readmeContentFallback = 'Please check README.md file directory.';
+const configTemplateFallback = 'Please check config-template.md file directory.';
+
+export const metadata: ConnectorMetadata = {
+  id: 'alipay',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with Alipay',
+    'zh-CN': '支付宝登录',
+  },
+  // TODO: add the real logo URL (LOG-1823)
+  logo: './logo.png',
+  description: {
+    en: 'Sign In with Alipay',
+    'zh-CN': '支付宝登录',
+  },
+  readme: existsSync(pathToReadmeFile)
+    ? readFileSync(pathToReadmeFile, 'utf8')
+    : readmeContentFallback,
+  configTemplate: existsSync(pathToConfigTemplate)
+    ? readFileSync(pathToConfigTemplate, 'utf-8')
+    : configTemplateFallback,
+};
+
+const alipayConfigGuard = z.object({
+  appId: z.string(),
+  privateKey: z.string(),
+  signType: z.enum(alipaySigningAlgorithms),
+});
+
+export type AlipayConfig = z.infer<typeof alipayConfigGuard>;
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  const result = alipayConfigGuard.safeParse(config);
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+  }
+};
+
+// Reference: https://github.com/alipay/alipay-sdk-nodejs-all/blob/10d78e0adc7f310d5b07567ce7e4c13a3f6c768f/lib/util.ts
+export const signingPamameters = (
+  parameters: AlipayConfig & Record<string, string>
+): Record<string, string> => {
+  const { biz_content: bizContent, privateKey, ...signParameters } = parameters;
+
+  if (bizContent) {
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    signParameters.biz_content = JSON.stringify(snakeCaseKeys(JSON.parse(bizContent)));
+  }
+
+  const decamelizeParameters = snakeCaseKeys(signParameters);
+
+  // eslint-disable-next-line @silverhand/fp/no-mutating-methods
+  const sortedParametersAsString = Object.keys(decamelizeParameters)
+    .sort()
+    .map((key) => {
+      const data = decamelizeParameters[key];
+
+      // Supported Encodings can be found at https://github.com/ashtuchkin/iconv-lite/wiki/Supported-Encodings
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      return data ? `${key}=${iconv.encode(data, signParameters.charset ?? 'UTF8')}` : '';
+    })
+    .filter((keyValueString) => keyValueString)
+    .join('&');
+
+  const sign = crypto
+    .createSign(alipaySigningAlgorithmMapping[signParameters.signType])
+    .update(sortedParametersAsString, 'utf8')
+    .sign(privateKey, 'base64');
+
+  return { ...decamelizeParameters, sign };
+};
+
+export const getAuthorizationUri: GetAuthorizationUri = async (redirectUri, state) => {
+  const { appId: app_id } = await getConnectorConfig<AlipayConfig>(metadata.id);
+
+  return `${authorizationEndpoint}?${stringify({
+    app_id,
+    redirect_uri: encodeURI(redirectUri), // The variable `redirectUri` should match {appId, appSecret}
```

Comment:
nit: const redirect_uri = encode(redirectUrl)

Reply:
`encodeURIComponent`?

Reply:
[LOG-2180](https://linear.app/silverhand/issue/LOG-2180/check-social-connectors-redirect-uri-logics) added to track this. Seems that `redirectUri` has already been encoded before being used as the input argument of getAuthorizationUri().